### PR TITLE
Add test for relative seconds, and rename conditions

### PR DESCRIPTION
**Pull request by matt-o-how**
This PR adds a new test for relative time, renames AGG_SIG to AGG_SIG_UNSAFE, and renames the time/height conditions to use "absolute" and "relative" naming conditions

### CODE REVIEW

**Comment by richardkiss** [comment]
Seems okay. I thought we'd talked about renaming the conditions `SECONDS_NOW` to `SECONDS_ABSOLUTE` and `SECONDS_AGE` to `SECONDS_RELATIVE`?

File: tests/blockchain/test_blockchain_transactions.py [hunk 1]
```diff
@@ -842,7 +842,67 @@ async def test_assert_block_age_exceeds(self, two_nodes):
...
+            await full_node_api_1.full_node.respond_block(full_node_protocol.RespondBlock(block))
+
+        # Coinbase that gets spent
+        block1 = blocks[2]
+        spend_coin_block_1 = None
+        for coin in list(block1.get_included_reward_coins()):
+            if coin.puzzle_hash == coinbase_puzzlehash:
+                spend_coin_block_1 = coin
+
+        # This condition requires block1 coinbase to be spent after 30 seconds from now
```
Comment (mariano54) [inline tests/blockchain/test_blockchain_transactions.py:869]: Do you mean 300 seconds in the comment?

File: tests/blockchain/test_blockchain_transactions.py [hunk 2]
```diff
@@ -842,7 +842,67 @@ async def test_assert_block_age_exceeds(self, two_nodes):
...
+        invalid_new_blocks = bt.get_consecutive_blocks(
+            1,
+            blocks,
+            farmer_reward_puzzle_hash=coinbase_puzzlehash,
+            transaction_data=block1_spend_bundle,
+            time_per_block=20,
+            guarantee_transaction_block=True,
+        )
+
+        # Try to validate that block before 30 sec
```
Comment (mariano54) [inline tests/blockchain/test_blockchain_transactions.py:887]: 300 or 30?

File: tests/blockchain/test_blockchain_transactions.py (hunk 1 above)
Comment (matt-o-how) [reply tests/blockchain/test_blockchain_transactions.py]: Changed to 300

File: tests/blockchain/test_blockchain_transactions.py (hunk 2 above)
Comment (matt-o-how) [reply tests/blockchain/test_blockchain_transactions.py]: Fixed

File: chia/full_node/mempool_check_conditions.py [hunk 3]
```diff
@@ -65,11 +65,11 @@ def mempool_assert_block_age_exceeds(
     except ValueError:
         return Err.INVALID_CONDITION
     if prev_transaction_block_height < block_index_exceeds_this:
-        return Err.ASSERT_HEIGHT_AGE_EXCEEDS_FAILED
+        return Err.ASSERT_HEIGHT_RELATIVE_FAILED
```
Comment (richardkiss) [inline chia/full_node/mempool_check_conditions.py:68]: Shouldn't this be `<=`?

File: chia/full_node/mempool_check_conditions.py [hunk 4]
```diff
@@ -49,11 +49,11 @@ def mempool_assert_block_index_exceeds(
     except ValueError:
         return Err.INVALID_CONDITION
     if prev_transaction_block_height < block_index_exceeds_this:
-        return Err.ASSERT_HEIGHT_NOW_EXCEEDS_FAILED
+        return Err.ASSERT_HEIGHT_ABSOLUTE_FAILED
```
Comment (richardkiss) [inline chia/full_node/mempool_check_conditions.py:52]: `<=` ?

File: chia/full_node/mempool_check_conditions.py (hunk 3 above)
Comment (richardkiss) [reply chia/full_node/mempool_check_conditions.py:68]: In the time-based ones, it's `<=`. And "exceeds" to me suggests "strictly exceeds" (ie. 4 does not exceed 4, but 5 does).

File: chia/util/errors.py [hunk 5]
```diff
@@ -127,7 +127,7 @@ class Err(Enum):
     TOO_MANY_BLOCKS = 102
     INVALID_CC_CHALLENGE = 103
     INVALID_PREFARM = 104
-    ASSERT_SECONDS_AGE_EXCEEDS_FAILED = 105
+    ASSERT_SECONDS_RELATIVE_FAILED = 105
```
Comment (richardkiss) [inline chia/util/errors.py:130]: The fact that this is not near the others is a tad embarrassing.

File: chia/wallet/puzzles/p2_m_of_n_delegate_direct.clvm [hunk 6]
```diff
@@ -54,7 +54,7 @@
 
     (defun add_aggsig (public_key_list hash delegated_puzzle solution)
         (if public_key_list
-            (c (list AGG_SIG (f public_key_list) hash)
+            (c (list AGG_SIG_UNSAFE (f public_key_list) hash)
```
Comment (richardkiss) [inline chia/wallet/puzzles/p2_m_of_n_delegate_direct.clvm:57]: Pretty sure we want `AGG_SIG_ME` here.

File: chia/wallet/puzzles/condition_codes.clvm [hunk 7]
```diff
@@ -29,10 +29,10 @@
     ; the conditions below ensure that we're "far enough" in the future
 
     ; wall-clock time
-    (defconstant ASSERT_SECONDS_AGE_EXCEEDS 80)
-    (defconstant ASSERT_SECONDS_NOW_EXCEEDS 81)
+    (defconstant ASSERT_SECONDS_RELATIVE 80)
```
Comment (richardkiss) [inline chia/wallet/puzzles/condition_codes.clvm:32]: When I see `ASSERT_SECONDS_RELATIVE 500` it's a lot less clear than `ASSERT_SECONDS_RELATIVE_EXCEEDS 500`, so I'm not really crazy about the name changes dropping "exceeds". Now there's no verb.

Comment (richardkiss) [reply chia/wallet/puzzles/condition_codes.clvm:32]: That's probably why it was `AGE` & `NOW` before, to keep the name from being too long without losing any semantics.

**Comment by wjblanke** [comment]
merged with richards OK

**Pull request merged**